### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by DemiMarie-parity**
The hard work was done by `cargo upgrade`.  Split out by request of
@niklasad1.

### CODE REVIEW

File: ethereum-types/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ethereum-types"
-version = "0.8.0"
+version = "0.8.1"
```
Comment (niklasad1) [inline ethereum-types/Cargo.toml]: this is just a `dev-dependency` change so no need to bump the version number

File: uint/tests/uint_tests.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ use core::u64::MAX;
 use core::str::FromStr;
 use core::convert::TryInto;
 use uint::{FromDecStrErr};
+extern crate rand;
```
Comment (debris) [inline uint/tests/uint_tests.rs]: `extern crate` is not needed if Cargo.toml `edition` is set to `2018`

**Comment by DemiMarie-parity** [comment]
@debris The reason to bump the minor and patch versions is to ensure that we are using a version that has all of the features and bug-fixes we need.  For example, `hyper 0.12.35` fixes a security vulnerability (HTTP request smuggling).

File: uint/src/uint.rs [hunk 3]
```diff
@@ -1577,7 +1580,8 @@ macro_rules! impl_quickcheck_arbitrary_for_uint {
 			fn arbitrary<G: $crate::quickcheck::Gen>(g: &mut G) -> Self {
 				let mut res = [0u8; $n_bytes];
 
-				let p = g.next_f64();
+				use $crate::rand::Rng;
```
Comment (niklasad1) [inline uint/src/uint.rs]: Can you explain this change?

It seems unnecessary and requires to have direct dendepency to `rand`, see https://docs.rs/quickcheck/0.9.0/quickcheck/trait.Gen.html.

File: primitive-types/impls/serde/Cargo.toml [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-version = "0.2.1"
+version = "0.2.2"
 authors = ["Parity Technologies <admin@parity.io>"]
 license = "Apache-2.0/MIT"
 homepage = "https://github.com/paritytech/parity-common"
 description = "Serde serialization support for uint and fixed hash."
 
 [dependencies]
-serde = "1.0"
+serde = "1.0.101"
```
Comment (ordian) [inline primitive-types/impls/serde/Cargo.toml]: let's not be too restrictive and revert this change

File: uint/Cargo.toml [hunk 5]
```diff
@@ -4,20 +4,21 @@ homepage = "http://parity.io"
...
 [dependencies]
-byteorder = { version = "1", default-features = false }
-rustc-hex = { version = "2.0", default-features = false }
-quickcheck = { version = "0.6", optional = true }
-crunchy = { version = "0.2", default-features = true }
+byteorder = { version = "1.3.2", default-features = false }
+rustc-hex = { version = "2.0.1", default-features = false }
+quickcheck = { version = "0.9.0", optional = true }
+crunchy = { version = "0.2.2", default-features = false }
+rand = { version = "0.7.2", default-features = false }
```
Comment (ordian) [inline uint/Cargo.toml]: I'd like to avoid dependency on rand, it seems unnecessary, we could use `next_u32` to generate 10/10/80% chances for quickcheck.

File: uint/fuzz/Cargo.toml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 name = "uint-fuzz"
 description = "Fuzzers for uint algorithms"
 publish = false
-version = "0.1.0"
+version = "0.1.1"
```
Comment (ordian) [inline uint/fuzz/Cargo.toml]: no need for a bump (not published)
```suggestion
version = "0.1.0"
```

File: uint/Cargo.toml [hunk 7]
```diff
@@ -4,20 +4,21 @@ homepage = "http://parity.io"
...
 edition = "2018"
 
 [dependencies]
-byteorder = { version = "1", default-features = false }
-rustc-hex = { version = "2.0", default-features = false }
-quickcheck = { version = "0.6", optional = true }
-crunchy = { version = "0.2", default-features = true }
+byteorder = { version = "1.3.2", default-features = false }
+rustc-hex = { version = "2.0.1", default-features = false }
+quickcheck = { version = "0.9.0", optional = true }
```
Comment (ordian) [inline uint/Cargo.toml]: this is technically a breaking change

File: primitive-types/impls/rlp/Cargo.toml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "impl-rlp"
-version = "0.2.0"
+version = "0.2.1"
```
Comment (ordian) [inline primitive-types/impls/rlp/Cargo.toml]: no need for a bump
```suggestion
version = "0.2.0"
```

File: parity-path/Cargo.toml [hunk 9]
```diff
@@ -1,11 +1,11 @@
 [package]
 name = "parity-path"
-version = "0.1.2"
+version = "0.1.3"
```
Comment (ordian) [inline parity-path/Cargo.toml]: no need for a bump
```suggestion
version = "0.1.2"
```

File: kvdb-memorydb/Cargo.toml [hunk 10]
```diff
@@ -1,12 +1,12 @@
 [package]
 name = "kvdb-memorydb"
-version = "0.1.2"
+version = "0.1.3"
```
Comment (ordian) [inline kvdb-memorydb/Cargo.toml]: no need for a bump
```suggestion
version = "0.1.2"
```

File: fixed-hash/Cargo.toml [hunk 11]
```diff
@@ -14,17 +14,17 @@ edition = "2018"
...
-rand = { version = "0.7", optional = true, default-features = false }
-rustc-hex = { version = "2.0", optional = true, default-features = false }
-quickcheck = { version = "0.9", optional = true }
-byteorder = { version = "1.2", optional = true, default-features = false }
-static_assertions = "0.3"
+rand = { version = "0.7.2", optional = true, default-features = false }
+rustc-hex = { version = "2.0.1", optional = true, default-features = false }
+quickcheck = { version = "0.9.0", optional = true }
+byteorder = { version = "1.3.2", optional = true, default-features = false }
+static_assertions = "1.0.0"
```
Comment (ordian) [inline fixed-hash/Cargo.toml]: technically a breaking change

File: contract-address/Cargo.toml [hunk 12]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "contract-address"
-version = "0.3.0"
+version = "0.3.1"
```
Comment (ordian) [inline contract-address/Cargo.toml]: no need for a bump
```suggestion
version = "0.3.0"
```

File: uint/Cargo.toml [hunk 13]
```diff
@@ -4,20 +4,21 @@ homepage = "http://parity.io"
...
+crunchy = { version = "0.2.2", default-features = false }
+rand = { version = "0.7.2", default-features = false }
 
 [features]
 default = ["std"]
-std = ["byteorder/std", "rustc-hex/std", "crunchy/std"]
+std = ["byteorder/std", "rustc-hex/std", "crunchy/std", "rand/std"]
```
Comment (niklasad1) [inline uint/Cargo.toml]: well, if we agree to keep `rand` it should not be an `std` feature because it is only used by `quickcheck`

File: primitive-types/impls/serde/Cargo.toml (hunk 4 above)
Comment (DemiMarie-parity) [reply primitive-types/impls/serde/Cargo.toml]: I see your point, but I also wonder why someone would want to use an older patch release of `serde`.  `1.0.101` also matches any higher minor or patch version.

File: uint/Cargo.toml (hunk 7 above)
Comment (DemiMarie-parity) [reply uint/Cargo.toml]: @ordian are you referring to bumping `quickcheck`?

File: primitive-types/impls/serde/Cargo.toml (hunk 4 above)
Comment (ordian) [reply primitive-types/impls/serde/Cargo.toml]: Serde doesn't follow semver though and makes patch releases for new features as well, so this doesn't state which features of serde we rely on. On the other hand, I don't see why anyone would not want the latest version.

File: uint/Cargo.toml (hunk 5 above)
Comment (ordian) [reply uint/Cargo.toml]: Alternatively, make rand optional
```suggestion
rand = { version = "0.7.2", default-features = false, optional = true }
```
I guess that would require renaming `quickcheck` dependency and exposing a new feature `quickcheck = ["<quickcheck-renamed>", "rand"]`

File: primitive-types/impls/serde/Cargo.toml (hunk 4 above)
Comment (dvdplm) [reply primitive-types/impls/serde/Cargo.toml]: We don't have a well defined policy wrt older rustc versions other than: "compiles on stable at the time of writing (the code)".

Personally I don't mind being explicit about the version here.

**Comment by DemiMarie-parity** [comment]
I had to re-add the `rand` dependency.  Otherwise, the code did not compile :cry:.

**Comment by ordian** [comment]
@DemiMarie-parity `uint/tests/uint_tests.rs` part of the patch was not applied, hence CI is failing

File: fixed-hash/Cargo.toml (hunk 11 above)
Comment (DemiMarie-parity) [reply fixed-hash/Cargo.toml]: I know, but not a big one.

**Comment by dvdplm** [comment]
@DemiMarie-parity can you please take care of publishing the updated crates when you have a chance? (And don't forget to tag!)

**Comment by DemiMarie-parity** [comment]
@dvdplm Yes, if I have the proper permissions.  I will let you know if I need help.
